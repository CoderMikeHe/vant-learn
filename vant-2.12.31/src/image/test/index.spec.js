import { mount } from '../../../test';
import VanImage from '..';

test('click event', () => {
  const wrapper = mount(VanImage);

  wrapper.trigger('click');
  expect(wrapper.emitted('click')[0][0]).toBeTruthy();
  wrapper.destroy();
});

test('load event', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      src: 'https://img01.yzcdn.cn/vant/cat.jpeg',
    },
  });

  wrapper.find('img').trigger('load');

  expect(wrapper.emitted('load')[0][0]).toBeTruthy();
  expect(wrapper).toMatchSnapshot();

  wrapper.setProps({ src: '' });
  expect(wrapper).toMatchSnapshot();
});

test('error event', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      src: 'https://img01.yzcdn.cn/vant/cat.jpeg',
    },
  });

  wrapper.find('img').trigger('error');

  expect(wrapper.emitted('error')[0][0]).toBeTruthy();
});

test('lazy load', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      src: 'https://img01.yzcdn.cn/vant/cat.jpeg',
      lazyLoad: true,
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('lazy-load load event', (done) => {
  const wrapper = mount(VanImage, {
    propsData: {
      lazyLoad: true,
      src: 'https://img01.yzcdn.cn/vant/cat.jpeg',
    },
    mocks: {
      $Lazyload: {
        $on(eventName, hanlder) {
          if (eventName === 'loaded') {
            setTimeout(() => {
              hanlder({ el: null });
              hanlder({ el: wrapper.find('img').element });
              expect(wrapper.emitted('load').length).toEqual(1);
              expect(wrapper).toMatchSnapshot();
              wrapper.destroy();
            });
          }
        },
        $off() {
          done();
        },
      },
    },
  });
});

test('lazy-load error event', (done) => {
  const wrapper = mount(VanImage, {
    propsData: {
      lazyLoad: true,
    },
    mocks: {
      $Lazyload: {
        $on(eventName, hanlder) {
          if (eventName === 'error') {
            setTimeout(() => {
              hanlder({ el: null });
              hanlder({ el: wrapper.find('img').element });
              expect(wrapper.emitted('error').length).toEqual(1);
              expect(wrapper).toMatchSnapshot();
              wrapper.destroy();
            });
          }
        },
        $off() {
          done();
        },
      },
    },
  });
});

test('show-loading prop', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      showLoading: false,
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('show-error prop', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      showError: false,
      src: 'https://img01.yzcdn.cn/vant/cat.jpeg',
    },
  });

  wrapper.find('img').trigger('error');

  expect(wrapper).toMatchSnapshot();
});

test('error-icon prop', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      errorIcon: 'error',
      src: 'https://img01.yzcdn.cn/vant/cat.jpeg',
    },
  });

  wrapper.find('img').trigger('error');

  expect(wrapper).toMatchSnapshot();
});

test('loading-icon prop', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      loadingIcon: 'success',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('apply icon-prefix prop to error-icon', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      errorIcon: 'error',
      iconPrefix: 'my-icon',
      src: 'https://img01.yzcdn.cn/vant/cat.jpeg',
    },
  });

  wrapper.find('img').trigger('error');

  expect(wrapper).toMatchSnapshot();
});

test('apply icon-prefix prop to loading-icon', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      loadingIcon: 'success',
      iconPrefix: 'my-icon',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('radius prop', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      radius: 3,
      src: 'https://img01.yzcdn.cn/vant/cat.jpeg',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('default slot', () => {
  const wrapper = mount(VanImage, {
    propsData: {
      src: 'https://img01.yzcdn.cn/vant/cat.jpeg',
    },
    scopedSlots: {
      default: () => 'Custom Default',
    },
  });

  expect(wrapper).toMatchSnapshot();
});
