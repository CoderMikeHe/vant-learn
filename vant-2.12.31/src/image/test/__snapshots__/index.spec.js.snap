// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apply icon-prefix prop to error-icon 1`] = `
<div class="van-image">
  <div class="van-image__error"><i class="my-icon my-icon-error van-image__error-icon">
      <!----></i></div>
</div>
`;

exports[`apply icon-prefix prop to loading-icon 1`] = `
<div class="van-image"><img class="van-image__img">
  <div class="van-image__loading"><i class="my-icon my-icon-success van-image__loading-icon">
      <!----></i></div>
</div>
`;

exports[`default slot 1`] = `
<div class="van-image"><img src="https://img01.yzcdn.cn/vant/cat.jpeg" class="van-image__img">
  <div class="van-image__loading"><i class="van-icon van-icon-photo van-image__loading-icon">
      <!----></i></div>Custom Default
</div>
`;

exports[`error-icon prop 1`] = `
<div class="van-image">
  <div class="van-image__error"><i class="van-icon van-icon-error van-image__error-icon">
      <!----></i></div>
</div>
`;

exports[`lazy load 1`] = `
<div class="van-image"><img class="van-image__img">
  <div class="van-image__loading"><i class="van-icon van-icon-photo van-image__loading-icon">
      <!----></i></div>
</div>
`;

exports[`lazy-load error event 1`] = `
<div class="van-image">
  <div class="van-image__error"><i class="van-icon van-icon-photo-fail van-image__error-icon">
      <!----></i></div>
</div>
`;

exports[`lazy-load load event 1`] = `<div class="van-image"><img class="van-image__img"></div>`;

exports[`load event 1`] = `<div class="van-image"><img src="https://img01.yzcdn.cn/vant/cat.jpeg" class="van-image__img"></div>`;

exports[`load event 2`] = `
<div class="van-image"><img src="" class="van-image__img">
  <div class="van-image__loading"><i class="van-icon van-icon-photo van-image__loading-icon">
      <!----></i></div>
</div>
`;

exports[`loading-icon prop 1`] = `
<div class="van-image"><img class="van-image__img">
  <div class="van-image__loading"><i class="van-icon van-icon-success van-image__loading-icon">
      <!----></i></div>
</div>
`;

exports[`radius prop 1`] = `
<div class="van-image" style="overflow: hidden; border-radius: 3px;"><img src="https://img01.yzcdn.cn/vant/cat.jpeg" class="van-image__img">
  <div class="van-image__loading"><i class="van-icon van-icon-photo van-image__loading-icon">
      <!----></i></div>
</div>
`;

exports[`show-error prop 1`] = `<div class="van-image"></div>`;

exports[`show-loading prop 1`] = `<div class="van-image"><img class="van-image__img"></div>`;
